### DISCUSSION TITLE
How can I use the same lib with different version in different packages?

### DISCUSSION THREAD

**Discussion by manzoliric** [pnpm/pnpm#6104] [Q&A]
Hello!

I have been working with monorepo for some months using Nuxt-2 for all my packages, but now, I want to create a new package using Nuxt-3!

I searched about hoist feature, but I can't make this work!

I searched and I found a similar problem in this repository https://github.com/Suniron/monorepo-nuxt2-nuxt3-problem, I have the same problem and the same structure.

**Comment by Suniron** [comment]
I think the problem can be fixed by editing the root `.npmrc` file but I dont really understand how 🤷‍♂️.

**Comment by iobuhov** [comment]
I checked your monorepo, and I see that you are using
```
shamefully-hoist=true
```
Which is probably why you have troubles.
Can you try this settings?
```
node-linker=isolated
hoist=false
```
Then you need to remove all `node_modules`, remove `pnpm-lock.yaml` and do clean `pnpm install`

**Reply by manzoliric** [reply]
I tried this setting, but didn't work.
`ERROR  Cannot find package '@nuxt/kit'`
![Screenshot 2023-02-20 at 09 33 30](https://user-images.githubusercontent.com/26907629/220109230-96730501-d533-4fc4-b7a8-27b106639325.png)

I checked all `node_modules` and in `nuxt2` project there are `@nuxt`, `@nuxtjs` and `nuxt` folders, but in `nuxt3` there is one single `nuxt` folder, maybe `nuxt3` needs a `@nuxt` but i don't know why pnpm don't created this folder.

**Reply by iobuhov** [reply]
```
❯ npm info nuxi dependencies
{ fsevents: '~2.3.2' }
```
`nuxi` is broken -- it doesn't list it's dependencies "explicitly". So this is not a `pnpm` problem.

In this case `@nuxt/kit` is phantom dependency and you need to use [public-hoist-pattern](https://pnpm.io/npmrc#public-hoist-pattern)
```
public-hoist-pattern[]=@nuxt/kit
```
And do it until all "phantom" deps are listed as "public". (all this while your `hoist=false`)

**Comment by Suniron** [comment]
Any update about this problem ? 😪.

I'm stucked since I can't build the nuxt 3 app in a monorepo with nuxt2/nuxt3 😢😢

**Comment by marc-tech** [comment]
hello, do you have any news or boilerplate that mix nuxt 2 and nuxt 3 ?
